rust/noosphere/src/ffi/petname.rs

```diff
@@ -0,0 +1,47 @@
+#![allow(unused_variables)]
+
+use safer_ffi::prelude::*;
+
+use crate::ffi::NsError;
+
+use super::NsNoosphereContext;
+
+#[ffi_export]
+/// Get the DID that is assigned to the provided petname; note that the DID is
+/// the ID of the sphere, but in order to read the sphere you must resolve the
+/// DID to a CID, which tells you the version of the sphere to read.
```

Comment:
A bit challenging to parse, separating the whats from the types; suggestion: "Get the identity as a DID that is assigned to the provided petname. To get the address associated with the petname, use `ns_sphere_petname_resolve()`."

The words "assigned to" and "associated with" suggests maybe we need a consistent name for this `(did, name)=>cid` object ("link"?) rather than "petname" being the core Thing here.

Reply:
You have the DID, yes. But then you need to resolve the DID to a CID. From there, the CID can be split into three parts: XID, YID and ZID. And with any combination of YID+ZID+DID or XID+YID+CID, you can derive the BID....

:D